deployment/postgres-init-db/sql/functions/hasura/plan_migration_functions.sql

```diff
@@ -0,0 +1,87 @@
+create table hasura.migrate_plan_to_model_return_value(result text);
+create function hasura.migrate_plan_to_model(_plan_id integer, _new_model_id integer, hasura_session json)
+  returns hasura.migrate_plan_to_model_return_value
+  volatile
+  language plpgsql as $$
+declare
+  _requester_username text;
+  _function_permission permissions.permission;
+  _open_merge_count integer;
+begin
+  _requester_username := (hasura_session ->> 'x-hasura-user-id');
+  _function_permission := permissions.get_function_permissions('migrate_plan_to_model', hasura_session);
+  perform permissions.raise_if_plan_merge_permission('migrate_plan_to_model', _function_permission);
```

Comment:
Major: This currently does not check the permissions, just checks that the permission isn't a plan merging-specific key. The following code checks the permission:
```suggestion
  perform permissions.raise_if_plan_merge_permission('migrate_plan_to_model', _function_permission);
    if not _function_permission = 'NO_CHECK' then
      call permissions.check_general_permissions('migrate_plan_to_model', _function_permission, _plan_id, _requester_username);
    end if;
```

Reply:
Thanks, updated in ea5fdc94d74554be63cd2417e642012851e439cc.